src/main/java/paymelah/logic/parser/SortCommandParser.java

```diff
@@ -0,0 +1,81 @@
+package paymelah.logic.parser;
+
+import static paymelah.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static paymelah.logic.parser.CliSyntax.PREFIX_MONEY;
+import static paymelah.logic.parser.CliSyntax.PREFIX_NAME;
+
+import java.util.Comparator;
+
+import paymelah.logic.commands.SortCommand;
+import paymelah.logic.parser.exceptions.ParseException;
+import paymelah.model.person.NameComparator;
+import paymelah.model.person.Person;
+import paymelah.model.person.TotalDebtAmountComparator;
+
+/**
+ * Parses input arguments and creates a new SortCommand object
+ */
+public class SortCommandParser implements Parser<SortCommand> {
+
+    public static final String CRITERIA_MONEY = "total amount owed";
+    public static final String CRITERIA_NAME = "name";
+    public static final String ORDER_ASCENDING = "ascending order";
+    public static final String ORDER_DESCENDING = "descending order";
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the SortCommand
+     * and returns an SortCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public SortCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_MONEY);
+
+        if ((!ParserUtil.arePrefixesPresent(argMultimap, PREFIX_NAME)
+                && (!ParserUtil.arePrefixesPresent(argMultimap, PREFIX_MONEY)))
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+        }
+
```

Comment:
Should we throw a ParseException if the user includes both n/ and m/ prefixes, even if only the last one is utilised?

Reply:
Good catch, will add checks for that